project/project_test.go

```diff
@@ -0,0 +1,868 @@
+	p := &mocks.MockClientService{}
+
+	c := &client.Harbor{
+		Products:  p,
+		Transport: nil,
+	}
+
+	cl := NewClient(c, authInfo)
+
+	ctx := context.Background()
+
+	getProjectParams := &products.GetProjectsParams{
+		Name:    &pReq.ProjectName,
+		Context: ctx,
+	}
+
+	p.On("GetProjects", getProjectParams, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
+		Return(&products.GetProjectsOK{
+			Payload: []*model.Project{{Name: exampleProject}}}, nil)
+
+	_, err := cl.GetProject(ctx, exampleProject)
+
+	assert.NoError(t, err)
+
+	p.AssertExpectations(t)
+}
+
+func TestRESTClient_ListProjects(t *testing.T) {
+	p := &mocks.MockClientService{}
+
+	c := &client.Harbor{
+		Products:  p,
+		Transport: nil,
+	}
+
+	cl := NewClient(c, authInfo)
+
+	ctx := context.Background()
+
+	getProjectParams := &products.GetProjectsParams{
+		Name:    &pReq.ProjectName,
+		Context: ctx,
+	}
+
+	p.On("GetProjects", getProjectParams, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
+		Return(&products.GetProjectsOK{
+			Payload: []*model.Project{{Name: exampleProject}}}, nil)
+
+	_, err := cl.ListProjects(ctx, exampleProject)
+
+	assert.NoError(t, err)
+
+	p.AssertExpectations(t)
+}
+
+func TestRESTClient_UpdateProject(t *testing.T) {
+	p := &mocks.MockClientService{}
+
+	c := &client.Harbor{
+		Products:  p,
+		Transport: nil,
+	}
+
+	cl := NewClient(c, authInfo)
+
+	ctx := context.Background()
+
+	getProjectParams := &products.GetProjectsParams{
+		Name:    &pReq.ProjectName,
+		Context: ctx,
+	}
+
+	putProjectParams := &products.PutProjectsProjectIDParams{
+		Project:   pReq,
+		ProjectID: exampleProjectID,
+		Context:   ctx,
+	}
+
+	p.On("GetProjects", getProjectParams, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
+		Return(&products.GetProjectsOK{
+			Payload: []*model.Project{{Name: exampleProject}}}, nil)
+
+	p.On("GetProjects", getProjectParams, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
+		Return(&products.GetProjectsOK{
+			Payload: []*model.Project{{Name: exampleProject}}}, nil)
+
+	p.On("PutProjectsProjectID", putProjectParams, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
+		Return(&products.PutProjectsProjectIDOK{}, nil)
+
+	project, err := cl.GetProject(ctx, exampleProject)
+
+	assert.NoError(t, err)
+
+	err = cl.UpdateProject(ctx, project, int(exampleCountLimit), int(exampleStorageLimit))
+
+	assert.NoError(t, err)
+
+	p.AssertExpectations(t)
+}
+
+func TestRESTClient_UpdateProject_IDMismatch(t *testing.T) {
+	p := &mocks.MockClientService{}
+
+	c := &client.Harbor{
+		Products:  p,
+		Transport: nil,
+	}
+
+	cl := NewClient(c, authInfo)
+
+	ctx := context.Background()
+
+	getProjectParams := &products.GetProjectsParams{
+		Name:    &pReq.ProjectName,
+		Context: ctx,
+	}
+
+	p.On("GetProjects", getProjectParams, mock.AnythingOfType("runtime.ClientAuthInfoWriterFunc")).
+		Return(&products.GetProjectsOK{
+			Payload: []*model.Project{{Name: exampleProject}}}, &ErrProjectMismatch{})
```

Comment:
Payload should be `nil`. The returned error type from `GetProjects` is usually something of `*runtime.APIError`